src/main/java/future/phase2/offlinetoonlinebazaar/service/Impl/RoleServiceImpl.java

```diff
@@ -0,0 +1,56 @@
+package future.phase2.offlinetoonlinebazaar.service.Impl;
+
+import future.phase2.offlinetoonlinebazaar.exception.ResourceNotFoundException;
+import future.phase2.offlinetoonlinebazaar.generator.IdGenerator;
+import future.phase2.offlinetoonlinebazaar.model.dto.RoleDto;
+import future.phase2.offlinetoonlinebazaar.model.entity.Role;
+import future.phase2.offlinetoonlinebazaar.model.enumerator.ErrorCode;
+import future.phase2.offlinetoonlinebazaar.repository.RoleRepository;
+import future.phase2.offlinetoonlinebazaar.service.RoleService;
+import ma.glasnost.orika.MapperFacade;
+import ma.glasnost.orika.MapperFactory;
+import ma.glasnost.orika.impl.DefaultMapperFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class RoleServiceImpl implements RoleService {
+
+    @Autowired
+    IdGenerator idGenerator;
+
+    @Autowired
+    RoleRepository roleRepository;
+
+    private MapperFactory mapperFactory = new DefaultMapperFactory.Builder().build();
```

Comment:
ini dipindah ke class baru ya, jadiin Component, biar bisa di reuse di tmpat lain. jd ga declare berkali kali. contoh:

public class BeanMapper {

  private static MapperFactory mapperFactory = new DefaultMapperFactory.Builder().build();
  private static MapperFacade mapper = mapperFactory.getMapperFacade();

/**
   *
   * @param source object yg mau diconvert
   * @param clazz tipe class destination yg mau di convert
   * @param <S>
   * @param <C>
   * @return list of object
   * digunakan untuk mapping Object -> AnotherObject
   */
  public static <S, C> C map(S source, Class<C> clazz) {
    if (source == null) {
      return null;
    }
    return mapper.map(source, clazz);
  }

  /**
   *
   * @param source object yg mau diconvert
   * @param clazz tipe class destination yg mau di convert
   * @param <S>
   * @param <C>
   * @return list of object
   * digunakan untuk mapping Collection<Object> -> Collection<AnotherObject>
   */
  public static <S, C> List<C> mapAsList(Iterable<S> source, Class<C> clazz) {
    if (source == null) {
      return null;
    }
    return mapper.mapAsList(source, clazz);
  }
}

Reply:
Kalau private gimana cara makainya ntar bg?

Reply:
tinggal manggil nya by class. itu kan static, jadi 
Beanmapper.map(s,d)

Reply:
fungsinya sama seperti yg contoh abg buat?